databricks/koalas/datetime.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright (C) 2019 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Date/Time related functions on Koalas Series
+"""
+
+import functools
+from pyspark.sql.types import DateType, TimestampType, LongType, StringType
+
+from databricks.koalas.series import Series
+from databricks.koalas.utils import lazy_property
+
+import pyspark.sql.functions as F
+
+
+def defer_to_pandas(output_type):
```

Comment:
I guess these are reusable for other series functions? If so, I'd move these to `series.py` or somewhere more global.

Reply:
Makes sense. Will move to some shared place.

Reply:
You are copying here most of the code in `typedef.py` and you do not need any of that. I have not tried the following but it should work, let me know if it does not:

```py
class DateTimeMethods(object):
  ...
    def strftime(self, date_format) -> ks.Series:
       return pandas_wrap(lambda col: col.strftime(col, date_format), return_col=StringType())(self._data)
```

This is all that you should need. There is no need to think about the parameters because they get embeded in the closure.